src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs

```diff
@@ -192,7 +192,12 @@ public static void GetVectorAndItemType(string name, Type rawType, IEnumerable<A
             // Check if the itemType extracted from rawType is supported by ML.NET's type system.
             // It must be one of either ML.NET's pre-defined types or custom types registered by the user.
             else if (!itemType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(itemType, attributes))
-                throw Contracts.ExceptParam(nameof(rawType), "Could not determine an IDataView type for member {0}", name);
+            {
+                //This is a big hack. It gives helpful information when customer accidently define OnnxSequenceType the same as member type
+                if (DataViewTypeManager.Knows(typeof(IEnumerable<>).MakeGenericType(itemType), attributes))
+                    throw Contracts.ExceptParam(nameof(rawType), "Defined uncessary container type of member {0}", name);
```

Comment:
Is it possible to provide a slightly more useful error message? Please look at the end of the bug description for #4122. There is a suggestion for what the error message should look like.

Reply:
sure